### PR TITLE
[GOVCMSD8-448] Added composer drupal optimisations.

### PR DESCRIPTION
**Pull request by stooit**
Adds specific composer optimisations for Drupal (https://github.com/zaporylie/composer-drupal-optimizations) which greatly reduces memory and CPU usage during composer commands (e.g `composer update` and `composer require`).

More detail and benchmarking is available at the source GH page, but for context a `composer update --profile` before and after:

**Before**:
`[314.4MiB/178.09s] Memory usage: 314.37MiB (peak: 695.66MiB), time: 178.09s`

**After**:
`[259.1MiB/13.95s] Memory usage: 259.14MiB (peak: 289.01MiB), time: 13.95s`

### CODE REVIEW

**Comment by pandaskii** [comment]
> If no configuration is provided this package will provide sensible defaults based on the drupal/core version constraint in the root composer.json file. 

@srowlands 

With this package, do we still need "webflo/drupal-core-strict"?

https://github.com/govCMS/govCMS8/blob/1.x/composer-lagoon.json#L33

**Comment by simesy** [comment]
@pandaskii i think we do. This is just filtering out loose dependencies to old symfony things that are very out of scope for drupal.

To me this tool is fine, we can share the love in terms of improving the composer experience and removing it would not be hard.

**Comment by simesy** [comment]
@tobybellwood would it help if I did some before/after memory comparison for this? Or are we ready to mrge.

File: composer.json [hunk 1]
```diff
@@ -101,7 +101,8 @@
     },
     "require-dev": {
         "grasmash/drupal-security-warning": "^1.1",
-        "phing/phing": "^2.16"
+        "phing/phing": "^2.16",
+        "zaporylie/composer-drupal-optimizations": "^1.1"
```
Comment (tobybellwood) [inline composer.json]: @simesy  - any idea if there is a benefit/downside to this being in require-dev - all other examples I've seen have it in require to ensure it's called every time?

Comment (simesy) [reply composer.json]: yes I think it needs to be in require otherwise it won't always applly.

**Review by simesy — APPROVED** [review]

**Comment by tobybellwood** [comment]
Moving the package into require (from require-dev)

Before `[736.1MiB/103.43s] Memory usage: 736.12MiB (peak: 1522.56MiB), time: 103.43s`

After `[683.6MiB/55.95s] Memory usage: 683.58MiB (peak: 1163.58MiB), time: 55.95s`

**Review by fubarhouse — APPROVED** [review]

**Comment by simesy** [comment]
Worth noting here that using satis.govcms.gov.au is:
```
Memory usage: 13.58MiB (peak: 16.89MiB), time: 7.04s
```

**Pull request merged**